lib/client/ipam_block.go

```diff
@@ -209,10 +210,16 @@ func (b *allocationBlock) release(addresses []cnet.IP) ([]cnet.IP, map[string]in
 	}
 
 	// Release requested addresses.
-	for _, ordinal := range ordinals {
+	uniqueOrdinals := set.FromArray(ordinals)
```

Comment:
If I understand this right, I think we need to de-duplicate the addresses a bit further up in this function.

Namely, the loop above where we iterate over the addresses does reference counting which won't be accurate in the case where duplicate addresses are provided, since the reference will be double counted in the loop above. 

Essentially this line: https://github.com/projectcalico/libcalico-go/pull/990/files#diff-1c8ffa16b869346e7e10b73ea36a9f42R180

Since `attrIdx` is tied directly to the ordinal being iterated over.

Reply:
Yes good point, will dedupe the addresses as the first step.